### DISCUSSION TITLE
Update modern to AVX2

### DISCUSSION THREAD

**Discussion by dav1312** [official-stockfish/Stockfish#4665] [Polls]
https://github.com/official-stockfish/Stockfish/blob/e87e103ca994570d42f30f61f923986656a5df14/src/Makefile#L788
AVX2 was released on Intel CPUs 10 years ago and on AMD CPUs 8 years ago.
https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX2
# Intel
- Haswell processors (only Core and Xeon branded), Q2 2013
- Haswell E processors, Q3 2014
- Broadwell processors, Q4 2014
- Broadwell E processors, Q3 2016
- Skylake processors, Q3 2015
- Kaby Lake processors, Q3 2016 (ULV mobile)/Q1 2017 (desktop/mobile)
- Skylake-X processors, Q2 2017
- Coffee Lake processors, Q4 2017
- Cannon Lake processors, Q2 2018
- Cascade Lake processors, Q2 2019
- Ice Lake processors, Q3 2019
- Comet Lake processors, Q3 2019
- Tiger Lake (Core, Pentium and Celeron branded) processors, Q3 2020
- Rocket Lake processors, Q1 2021
- Alder Lake (Xeon, Core, Pentium and Celeron branded) processors, Q4 2021. Supported both in Golden Cove P-cores and Gracemont E-cores.
- Raptor Lake processors, Q4 2022
- Sapphire Rapids processors, Q1 2023
# AMD
- Excavator processor and newer, Q2 2015
- Zen processors, Q1 2017
- Zen+ processors, Q2 2018
- Zen 2 processors, Q3 2019
- Zen 3 processors, Q4 2020
- Zen 4 processors, Q4 2022

**Comment by vondele** [comment]
What about dropping the use of that x86-64-modern arch in any of our CI, releases, wiki examples, etc but just keep it for backwards compatibility (i.e. so the target is available, it may still be commonly referenced)?

In that case we can also update it, as it won't matter much.

**Comment by Sopel97** [comment]
I agree with vondele, we should move away from targets with vague and changing meaning completely.

**Comment by ppigazzini** [comment]
I'm to drop "x86-64-modern" everywhere, keep a temporary deprecated target to be dropped as well after N month(s).

**Comment by jhellis3** [comment]
I'd vote for renaming it popcnt.

**Comment by vondele** [comment]
It currently is an alias for `x86-64-sse41-popcnt` so I'd keep that name I think (we also have x86-64-sse3-popcnt).

But, my feeling is that most want to stop using / deprecating x86-64-modern as an arch specifier, and explicitly call the thing by its full name. On the download page we'll have to find a way to guide users to the appropriate binary (probably taking into account that most users don't want to be try multiple binaries that crash, btw).

**Comment by vondele** [comment]
see https://github.com/official-stockfish/Stockfish/pull/4691
